libs/utils/trace.py

```diff
@@ -81,7 +81,7 @@ def __init__(self, platform, data_dir, events,
                  plots_prefix=''):
 
         # The platform used to run the experiments
-        self.platform = platform
+        self.platform = platform or {}
```

Comment:
Good idea that could definitely mitigate the cost. Anyway I think let's keep it for a separate PR, here I'll just try to allow the first param to be None